web/cobrands/buckinghamshire/assets.js

```diff
@@ -0,0 +1,193 @@
+];
+// And these are roads they don't maintain.
+var non_bucks_types = [
+    "HE", // HW: HIGHWAYS ENGLAND
+    "HWOA", // OTHER AUTHORITY
+    "HWSA", // HW: Whole Street Asset
+    "P", // HW: PRIVATE
+];
+
+// We show roads that Bucks are and aren't responsible for, and display a
+// message to the user if they click something Bucks don't maintain.
+var types_to_show = bucks_types.concat(non_bucks_types);
+
+// Some road types we don't want to display at all.
+var types_to_hide = [
+    "11", // HW: BYWAY OPEN TO TRAFFIC
+    "12", // HW: FOOTPATH PROW
+    "13", // HW: BYWAY RESTRICTED
+    "14", // HW: BRIDLEWAY
+    "9", // HW: NO CARRIAGEWAY
+];
+
+var highways_style = new OpenLayers.Style({
+    fill: false,
+    strokeColor: "#5555FF",
+    strokeOpacity: 0.1,
+    strokeWidth: 7
+});
+
+function bucks_owns_feature(f) {
+    return f &&
+           f.attributes &&
+           f.attributes.feature_ty &&
+           bucks_types.indexOf(f.attributes.feature_ty) > -1;
+}
+
+function bucks_does_not_own_feature(f) {
+    return !bucks_owns_feature(f);
+}
+
+var rule_owned = new OpenLayers.Rule({
+    filter: new OpenLayers.Filter.FeatureId({
+        type: OpenLayers.Filter.Function,
+        evaluate: bucks_owns_feature
+    })
+});
+
+var rule_not_owned = new OpenLayers.Rule({
+    filter: new OpenLayers.Filter.FeatureId({
+        type: OpenLayers.Filter.Function,
+        evaluate: bucks_does_not_own_feature
+    }),
+    symbolizer: {
+        strokeColor: "#555555"
+    }
+});
+highways_style.addRules([rule_owned, rule_not_owned]);
+
+function show_responsibility_error(id) {
+    hide_responsibility_errors();
+    $("#js-bucks-responsibility").removeClass("hidden");
+    $("#js-bucks-responsibility .js-responsibility-message").addClass("hidden");
+    $(id).removeClass("hidden");
+}
+
+function hide_responsibility_errors() {
+    $("#js-bucks-responsibility").addClass("hidden");
+    $("#js-bucks-responsibility .js-responsibility-message").addClass("hidden");
+}
+
+function disable_report_form() {
+    $("#problem_form").hide();
+}
+
+function enable_report_form() {
+    $("#problem_form").show();
+}
+
+fixmystreet.assets.add($.extend(true, {}, defaults, {
+    http_options: {
+        params: {
+            TYPENAME: "Whole_Street"
+        }
+    },
+    stylemap: new OpenLayers.StyleMap({
+        'default': highways_style
+    }),
+    always_visible: true,
+    non_interactive: true,
+    road: true,
+    asset_item: 'road',
+    all_categories: true,
+    actions: {
+        found: function(layer, feature) {
+            if (bucks_types.indexOf(feature.attributes.feature_ty) != -1) {
+                hide_responsibility_errors();
+                enable_report_form();
+            } else {
+                // User has clicked a road that Bucks don't maintain.
+                show_responsibility_error("#js-not-bucks-road");
+                disable_report_form();
+            }
+        },
+
+        not_found: function(layer) {
```

Comment:
Should this function (and `found` above) change its behaviour according to which cobrand is in effect? I think it might be out of the scope of this PR to implement the "your report isn't on a Bucks road, so we won't send it to them, but it will be displayed on FMS.com" feature, but perhaps it should only be active if the current cobrand is Buckinghamshire.

Reply:
Yeah, I was wondering if it makes sense to work out how to have national asset behaviour and cobrand asset behaviour, although I need to work out a clean way to do that.